### DISCUSSION TITLE
Demand control on single split means anything?

### DISCUSSION THREAD

**Discussion by yufaxt** [revk/xESP32-Faikout#439] [Q&A] [answered]
Hi Community,

First of all many thanks for the discussions here and of course this great project. It really makes my A/C much more pleasant to use :)

I was wondering if demand control on a single unit means anything. I was playing with it a bit but not sure if it did anything.

what I was hoping to reach is a smoother auto mode. I noticed my Emura 3 will stop the fan completely when max is reached if coolback is higher than 0. as far as I understand the best practice with such inverter ACs is to let them run but rather using the minimal power required to maintain the constant temperature. However in this case if I set coolback to 0 the unit is cooling too much even on night mode, so that the reached temperature is well below the max. e.g max being 25 and env already at 24, daikin own sensor sees 24.5 and the temperature is set to 25 hence the daikin should be at minimal consumption yet env stays at 24 nevertheless. If I increase coolback env temp will go back up due to the daikin being virtually off. Ideally I should be able to find a way to keep it running but use even less power so the temperature can increase a bit. suggestions are welcomed.

btw I’m running faikin on the latest beta and seeing an issue with the auto controls on the web interface disappearing randomly. I than get them back by enabling nofaikinauto and disabling it again. is this only me?

Thanks in advance

**Comment by revk** [comment]
I don't know much about demand control, we just send the message to the a/c :-)

However, the auto controls disappearing is odd, I'll look.

**Reply by revk** [reply]
I think some of the BLE tests are confused, so I'll make a new beta in case that helps.

**Reply by MUN0X** [reply, marked as answer]
Demand control is basically a power limit that can be forced upon the outdoor unit. 
Its a dumb name imho, `power limit` would be much clearer.
Daikin markets it as [Demand contol decreases the power consumption to 70% or 40% when other larger appliances need to be switched on](https://www.daikin.co.uk/en_gb/products/product.html/BRC1E53A.html)

This is very useful for multisplits as they tend to have a large/powerful outdoor unit to accommodate all indoor units simultaneous.  However when only a single indoor unit is running it can be overwhelmed by
 the big outdoor unit. Demand control can be used to “overrule” the aggressive outdoor unit in these cases. I can imagine this can be used for single splits as well but there is probably less need to do so since in- and outdoor units will equally sized.

In my experience demand control does very little in cooling mode. The daikin “algorithm” seems very well optimized for cooling with the outdoor unit running at minimum compressor frequency once the setpoint is being approached. Even with a big 4-split in my case. In heating mode on the other hand, changing demand control can make a massive difference.

I dont think you should change coolback to achieve smoother control. Coolback is faikins way to turn the unit off and you shouldn’t take that away. Instead play with `thermostat`, `tpredict`, `tsample`, `temptrack`, `tempadjust` and `coolover` to get the desired behavior.

That being said, on/off cycles are not a huge deal for a single split since there are no other units that can bleed refrigerant.

**Reply by yufaxt** [reply]
Thank you very much for the detailed answer. Will try that and post findings

**Reply by yufaxt** [reply]
> I think some of the BLE tests are confused, so I'll make a new beta in case that helps.

I am running now on 2024-08-17T08:42:13 and I think I managed to spot when the auto mode controls disappear. after a restart they are there, but then as soon as I report env temp via mqtt (only env in the message), it is gone and the env temperature is added in the UI. The unit is off I just report the temperature from an external sensor every minute

**Reply by revk** [reply]
Ah, sending the MQTT messages for control remotes it as you are controlling remotely not using Faikin auto.

What are you sending, just temp? If so that may be the issue, as you are not sending a target. The command was designed originally for reference temp and target details. I can probably tweak that.

**Reply by revk** [reply]
That should not be new behaviour, but I have changed, only `target` in `control` sets as MQTT remote. Try next beta in a few minutes.

**Reply by yufaxt** [reply]
Correct! I could add target but then I complicate the rather simple to use target control over faikin UI/HA. the idea was to just let the Faikin know the env temp all the time, and nothing else.

For simplicity I just publish the message every minute, also when the AC is off (hence auto is also off). I don’t let auto mode also really turn the unit on and off automatically, I do it manually and only let it control unit when I turn the unit on.

I thought that should be a common use case as probably many want to get a temperature from a sensor positioned elsewhere (e.g my unit is positioned so high up, it always sees 2-3 degrees warmer than the actual temperature I feel)

**Reply by revk** [reply]
I have made a beta but may have to make a further change as not sure it will use your setting now.

**Reply by revk** [reply]
Actually it may be right, check the latest beta.

**Reply by yufaxt** [reply]
2024-08-18T07:53:50 Seems to work for me. Thanks for handling so quickly 🙏
